### PR TITLE
Add endpoint to get pins for an investigation

### LINKED ISSUE #56: Create endpoints for Pins
Useful endpoints could include:

- Get a pin by ID (include all fields)
- Get all pins associated with an Investigation (include minimal fields like ID + location data)

### PR DESCRIPTION
**Pull request by annehaley**
Resolves #56

### CODE REVIEW

File: atlascope/core/rest/endpoints/investigation_endpoints.py [hunk 1]
```diff
@@ -89,3 +90,9 @@ def permissions(self, request, pk=None):
         payload = InvestigationDetailSerializer(investigation).data
         payload = {k: v for k, v in payload.items() if k in ['owner', 'investigators', 'observers']}
         return Response(payload, status=status.HTTP_200_OK)
+
+    @object_permission_required(edit_access=True)
+    @action(detail=True, methods=['GET'])
+    def pins(self, request, pk=None):
+        payload = {str(pin.id): PinSerializer(pin).data for pin in self.get_object().pins.all()}
```
Comment (naglepuff) [inline atlascope/core/rest/endpoints/investigation_endpoints.py]: I think a list is preferable to a dictionary here. Maybe something like
`payload = PinSerializer(self.get_object().pins.all(), many=True).data`?

File: atlascope/core/rest/endpoints/investigation_endpoints.py [hunk 2]
```diff
@@ -89,3 +90,9 @@ def permissions(self, request, pk=None):
         payload = InvestigationDetailSerializer(investigation).data
         payload = {k: v for k, v in payload.items() if k in ['owner', 'investigators', 'observers']}
         return Response(payload, status=status.HTTP_200_OK)
+
+    @object_permission_required(edit_access=True)
+    @action(detail=True, methods=['GET'])
+    def pins(self, request, pk=None):
```
Comment (naglepuff) [inline atlascope/core/rest/endpoints/investigation_endpoints.py:97]: Could you add a `@swagger_auto_schema` wrapper here to clarify the response schema? Right now it just shows up as the `InvestigationDetail` on the swagger page.

**Review by naglepuff — CHANGES_REQUESTED** [review]

File: atlascope/core/rest/endpoints/investigation_endpoints.py (hunk 1 above)
Comment (annehaley) [reply atlascope/core/rest/endpoints/investigation_endpoints.py]: I like this a lot better, too. Changed in 722402a

File: atlascope/core/rest/endpoints/investigation_endpoints.py (hunk 2 above)
Comment (annehaley) [reply atlascope/core/rest/endpoints/investigation_endpoints.py:97]: Changed in 722402a

**Review by naglepuff — APPROVED** [review]

**Pull request merged**